File: Java/Problems/Sorting/SelectionSort.java
public class SelectionSort {

    public static void main(String[] args) {
        
        int[] arr = {64, 25, 12, 22, 11};
        
        int n = arr.length;

        // Total number of round (n-1)
        for (int i = 0; i < n-1; i++) {

            //find minimum of the next array from (i+i)
            int min_idx = i;
            for (int j = i+1; j < n; j++) {
                if (arr[j] < arr[min_idx])
                    min_idx = j;                
            }
        //swap the found element with the ith element
        int temp = arr[min_idx];
        arr[min_idx] = arr[i];
        arr[i] = temp; 
        }

        //print the aray
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
}

File: C++/Problems/Sorting/selectionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

void selectionSort(vector<int> v, int n)
{
	int i, j, min_idx;

	// One by one move boundary of unsorted subarray
	for (i = 0; i < n-1; i++)
	{
		// Find the minimum element in unsorted array
		min_idx = i;
		for (j = i+1; j < n; j++)
			if (v[j] < v[min_idx])
				min_idx = j;

		// Swap the found minimum element with the first element 
		if(min_idx != i)
			swap(v[min_idx], v[i]);
	}
}

//Function to print the data
void print(vector<int> v, int size)
{
	int i;
	for (i=0; i < size; i++)
		cout << v[i] << " ";
	cout << endl;
}

// Main Function
int main()
{
	vector<int> v = {64, 25, 12, 22, 11};
	int n = v.size();
	selectionSort(v, n);
	cout << "Sorted array: " << endl;
	print(v, n);
	return 0;
}
 